### PR TITLE
h5cpp: Correct checksum and build instructions

### LINKED ISSUE #20046: Installation issue: h5cpp has invalid checksum
### Steps to reproduce the issue

```console
$ spack install h5cpp
```

### Information on your system

* **Spack:** 0.16.0-32-4609a126b
* **Python:** 3.6.8
* **Platform:** linux-centos8-broadwell
* **Concretizer:** original

### Additional information

Invalid SHA256 Checksum

Checksum in spack package module: 42d0ca1aaff1ead8998a26d892a51c12b1b89023382f191dc438bd0fa4513455
Checksum upstream: 661ccc4d76e081afc73df71ef11d027837d92dd1089185f3650afcaec9d418ec

@eschnett

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by eschnett**
Closes https://github.com/spack/spack/issues/20046.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Any idea why the checksum changed? Is there any way we can verify that the new tarball doesn't contain a virus?

**Comment by eschnett** [comment]
I don't know why it changed. However, I recall that github repackages (or used to repackage?) the tarfiles from time to time, and that changes the checksums. I don't know whether there is a way to get immutable tarballs.

I notice that some repos have tarballs with a `releases/download` url, while others just have an `archive` url. Maybe there's a difference?

**Comment by adamjstewart** [comment]
The `archive` tarball is automatically generated, while the `releases/download` tarballs are user generated.

File: var/spack/repos/builtin/packages/h5cpp/package.py [hunk 1]
```diff
@@ -26,4 +27,7 @@ class H5cpp(CMakePackage):
     depends_on('mpi', when='+mpi')
 
     def cmake_args(self):
-        return ['-DH5CPP_BUILD_TESTS=OFF']
+        return ['-DHDF5_INCLUDE_DIRS=%s' % self.spec['hdf5'].prefix.include,
+                '-DHDF5_LIBRARIES=%s' % self.spec['hdf5'].prefix.lib,
+                '-DH5CPP_BUILD_TESTS=OFF',
+                ]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/h5cpp/package.py]:
```suggestion
        return [
            '-DHDF5_INCLUDE_DIRS=%s' % self.spec['hdf5'].headers.directories[0],
            '-DHDF5_LIBRARIES=%s' % self.spec['hdf5'].libs.directories[0],
            '-DH5CPP_BUILD_TESTS=OFF',
        ]
```
Libraries may be in `prefix.lib` or `prefix.lib64`.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/h5cpp/package.py:31: [E501] line too long (80 > 79 characters)
```

**Comment by eschnett** [comment]
`style and docs` failure is unrelated.

**Pull request merged**